docs: update SSH command format to use suffix

matifali opened the pull request:
Refactor the workspace SSH command syntax across the project to use the "workspace.coder" format instead of "coder.workspace". This standardizes the SSH host entries for better consistency and clarity.

This is a follow-up from #17445 and recommends using the suffix-based format for all new Coder versions.

<img width="418" alt="image" src="https://github.com/user-attachments/assets/3893f840-9ce1-4803-a013-736068feb328" />

spikecurtis commented on site/src/modules/resources/SSHButton/SSHButton.tsx:
```diff
@@ -56,7 +56,7 @@ export const AgentSSHButton: FC<AgentSSHButtonProps> = ({
 						<SSHStep
 							helpText="Connect to the agent:"
-							codeExample={`ssh ${sshPrefix}${workspaceName}.${agentName}`}
+							codeExample={`ssh ${workspaceName}.${agentName}.${sshSuffix}`}
```
```suggestion
							codeExample={`ssh ${agentName}.${workspaceName}.me.${sshSuffix}`}
```

This is the Coder Connect format, which I think is what we should be getting people used to using.

matifali replied:
Sure. I will update.

matifali replied:
If an admin visits a different user's workspace and tries to connect using the above command with `*.me.coder`, it will fail.
Should we either drop `*.me.coder` or replace it with the workspace owner's username?

matifali replied:
By adding a username
<img width="399" alt="image" src="https://github.com/user-attachments/assets/7333bf40-9b4a-47fc-9b9f-740b1170a60a" />
This is the view from an admin checking the workspace owned by `user1`.

spikecurtis commented on site/src/pages/WorkspacePage/WorkspaceReadyPage.tsx:
```diff
@@ -208,7 +208,7 @@ export const WorkspaceReadyPage: FC<WorkspaceReadyPageProps> = ({
 				hideVSCodeDesktopButton={featureVisibility.browser_only}
 				buildInfo={buildInfoQuery.data}
-				sshPrefix={sshPrefixQuery.data?.hostname_prefix}
+				sshSuffix={sshConfigQuery.data?.hostname_suffix}
```
One consequence of this change is that if operators have the prefix set to something other than `coder`, but haven't set the suffix, then their custom name won't be used in the call to action. It won't prevent the prefix from still working, but might be surprising to see `.coder` in the UI.

It can be their encouragement to set the new config option, but could generate some support tickets.

matifali replied:
I think we need to force-push here to promote the use of the suffix.

spikecurtis replied:
I agree that we should mark CODER_SSH_HOSTNAME_PREFIX as deprecated, but that can be a separate PR.

matifali replied:
Filed https://github.com/coder/coder/issues/18156

matifali committed:
Update SSH usage examples and documentation

Revise various parts of the CLI and documentation to reflect the new SSH connection format for workspaces. Update usage examples from the outdated "ssh coder.workspace" format to the new "ssh workspace.coder" format. Additionally, ensure correct formation in commands by adding workspace owner username and hostname suffix through recent data sources.
